app/models/case_log.rb

```diff
@@ -477,12 +477,16 @@ def set_derived_fields!
     (2..8).each do |idx|
       if public_send("age#{idx}") && public_send("age#{idx}") < 16
         self["ecstat#{idx}"] = 9
-      elsif public_send("ecstat#{idx}") == 9 && (public_send("age#{idx}").nil? || public_send("age#{idx}") >= 16)
+      elsif public_send("ecstat#{idx}") == 9 && (public_send("age#{idx}").nil? || public_send("age#{idx}") >= 16) && age_known?(idx)
```

Comment:
My only comment is that we could do some minor refactoring of this to make the full set of conditions easier to understand - like you've done with the `age_known?` method